client/subchannel.go

```diff
@@ -29,7 +29,22 @@ func (c *Channel) IsLedgerChannel() bool {
 
 // IsSubChannel returns whether the channel is a sub-channel.
 func (c *Channel) IsSubChannel() bool {
-	return c.Parent() != nil
+	return c.Parent() != nil && c.equalParticipants(c.Parent())
```

Comment:
Can't we actually read this from the params?

Reply:
You mean by a flag? Unfortunately not. The original idea was that the params don't contain that information and they still don't, even though they now contain flags for virtualChannel and ledgerChannel.

Reply:
So the flag is not guaranteed to match that information?

Reply:
Not sure what you mean. There is no subchannel flag.

Reply:
I thought _sub channel_ means, _not ledger channel_, and there's a flag for this in the params.